test/lib/flat_mutation_reader_assertions.hh

```diff
@@ -780,10 +844,12 @@ public:
         if (bool(mfopt)) {
             BOOST_FAIL(format("Expected end of stream, got {}", mutation_fragment_v2::printer(*_reader.schema(), *mfopt)));
         }
+        reset_rt();
```

Comment:
I would say the same comment applies here as to `produce_partition_end()` but in fact this method may be called after abruptly finishing consumption, so I don't think we can be that strict here. Maybe we can remember whether there was a `next_partition()` or `fast_forward_to()` call before `produces_end_of_stream()`, but maybe not worth the churn.

Reply:
I see you already reset the rt in `next_partition()` so maybe we can be strict here after all. Still not sure because of `fast_forward_to()`.

Reply:
I added the `reset_rt()` to `produces_end_of_stream()` exactly because some "abruptly finishing consumption" cases seemingly called for it (as in "I haven't found a more elegant way to make those tests pass")